wrapper/src/main/java/software/amazon/jdbc/plugin/dev/DeveloperConnectionPlugin.java

```diff
@@ -24,13 +24,15 @@
 import java.util.Set;
 import java.util.logging.Logger;
 import org.checkerframework.checker.nullness.qual.NonNull;
+import software.amazon.jdbc.AuthenticationConnectionPlugin;
 import software.amazon.jdbc.HostSpec;
 import software.amazon.jdbc.JdbcCallable;
 import software.amazon.jdbc.plugin.AbstractConnectionPlugin;
 import software.amazon.jdbc.util.StringUtils;
 import software.amazon.jdbc.util.WrapperUtils;
 
-public class DeveloperConnectionPlugin extends AbstractConnectionPlugin implements ExceptionSimulator {
+public class DeveloperConnectionPlugin extends AbstractConnectionPlugin
+    implements AuthenticationConnectionPlugin, ExceptionSimulator {
```

Comment:
Similar to my previous comment, it seems a bit misleading to label this as an Authentication plugin when it doesn't actually provide auth functionality. Can we similarly either remove this and explicitly check for this plugin in ConnectionPluginManager#makePluginChainFunc or do something like rename AuthenticationConnectionPlugin to ForceConnectPlugin